spec/requests/admin/site_texts_controller_spec.rb

```diff
@@ -58,6 +58,28 @@
         expect(JSON.parse(response.body)['extras']['has_more']).to be_truthy
       end
 
+      it 'works with pages' do
+        texts = Set.new
+
+        get "/admin/customize/site_texts.json", params: { q: 'e' }
+        JSON.parse(response.body)['site_texts'].each { |text| texts << text['id'] }
+        expect(texts.size).to eq(50)
+
+        get "/admin/customize/site_texts.json", params: { q: 'e' }
+        JSON.parse(response.body)['site_texts'].each { |text| texts << text['id'] }
+        expect(texts.size).to eq(100)
+      end
+
+      it 'works with locales' do
+        get "/admin/customize/site_texts.json", params: { q: 'yes_value', locale: 'en' }
+        value = obj['site_texts'].find { |text| text['id'] == 'js.yes_value' }['value']
+        expect(value).to eq('Yes')
```

Comment:
This test will fail if/when the translation changes. It's best to call `I18n.t()` with the `en` locale to make this test less fragile.